src/figuresClasses.ts

```diff
@@ -1,11 +1,94 @@
-export interface Figure {}
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+type Color = 'red' | 'green' | 'blue';
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+
+  getArea(): number;
+}
 
-export class Triangle implements Figure {}
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
```

Comment:
[CODE KNOWLEDGE] - You do not need to manually assign constructor parameters to class properties when using access modifiers in TypeScript. Since `color`, `a`, `b`, and `c` are declared public in the constructor, there is no need for manual assignment.